### ISSUE TITLE
Fix javadoc warnings

### ISSUE THREAD

**Issue by jroper** [playframework/playframework#4995]
Currently our build outputs around 300 warnings from javadoc, most because `@return` and `@param` directives are missing.  This should be fixed.

**Comment by jroper** [comment]
This issue doesn't have to be done all at once - a new contributor can just fix a few of the javadoc warnings and that would be much appreciated.

**Comment by marcospereira** [comment]
Also, since our [our contributors guice](contributor guidelines) suggest that we follow the [boy scout rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule), it could be a good idea that new PR fixes javadocs warnings of changed classes. :wink:

**Comment by eboto** [comment]
So I'm currently 80% of the way to fixing all scaladoc and javadoc warnings in the framework. This work has included the following:

* Adding `autoAPIMappings` in the default play sub-builds so their docs can talk to each other
* Providing sbt APIMappings to the scala standard library for each sub-build so their docs can reference the types in the standard library
* Hard-coding scaladoc links to java types using [the oracle java 8 javadoc base URL](https://docs.oracle.com/javase/8/docs/api/)
  * Since scaladoc can not reference javadocs directly
* Cleaning up the documentation for thousands of classes, methods, and arguments

I still have a few questions:
* Is it best to submit what I have now (which is already a grotesquely large changeset) or just finish up the odd 500 remaining javadoc errors (all but 1 scaladoc error is fixed already)
* Once this work is done, how do you feel about an option to make scaladoc / javadoc more strict as part of the build process, so we don't get to this stage again?

**Comment by jroper** [comment]
Hi @eboto, there's at least one other person who has done some work on this, so I'd suggest sending a PR with everything you've done sooner rather than later.

I definitely agree with making things more strict.  At some point I'd love to enable -Xfatal-warnings in the scala/scaladoc compile, though there are a few things we'll need to address before we do that, like how to deprecate APIs but then still use them in our code (since sometimes you have to still invoke deprecated methods in order to ensure other deprecated functionality still works).

**Comment by eboto** [comment]
@jroper you got it boss. That PR contains my best first shot. There's one scaladoc warning remaining and about 500 javadoc warnings (down from I would guess over a thousand).

As for deprecation, it would be great if you could tell scalac to ignore some deprecations the way you can in javac, but barring that I'm not sure of the solution =(

Would definitely appreciate someone going through the changeset with a fine tooth comb because my eyes did occasionally start to glaze over.

**Comment by wsargent** [comment]
Closing via #6383
